File: app/src/main/java/ru/eva/oriokslive/activities/schedule/SchedulerActivity.java
package ru.eva.oriokslive.activities.schedule;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

import ru.eva.oriokslive.R;
import ru.eva.oriokslive.adapters.SchedulerFragmentPagerAdapter;

public class SchedulerActivity extends AppCompatActivity implements ContractSchedulerActivity.View {

    private ViewPager viewPager;

    private ContractSchedulerActivity.Presenter mPresenter;
    private SchedulerFragmentPagerAdapter adapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scheduler);
        String group = getIntent().getStringExtra("group");

        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle(group);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        viewPager = findViewById(R.id.view_pager);
        TabLayout tabLayout = findViewById(R.id.sliding_tabs);
        tabLayout.setupWithViewPager(viewPager);
        adapter = new SchedulerFragmentPagerAdapter(getSupportFragmentManager());
        adapter.setGroup(group);

        mPresenter = new PresenterSchedulerActivity(this);
        mPresenter.setPagerAdapter(group);
    }

    @Override
    public void setPagerAdapter() {
        viewPager.setAdapter(adapter);
    }

    @Override
    public void showToast(String text) {
        Toast.makeText(this, text, Toast.LENGTH_LONG).show();
    }

    @Override
    public void setViewPagerToPosition(int position) {
        viewPager.setCurrentItem(position);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_date:
                mPresenter.setViewPagerToCurrentWeek();
                return true;
                default:
                    onBackPressed();
                    return true;
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main, menu);
        return super.onCreateOptionsMenu(menu);
    }
}

File: app/src/main/java/ru/eva/oriokslive/ui/activity/schedule/SchedulerActivity.kt
package ru.eva.oriokslive.ui.activity.schedule

import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import androidx.activity.viewModels
import com.google.android.material.tabs.TabLayoutMediator
import dagger.hilt.android.AndroidEntryPoint
import ru.eva.oriokslive.R
import ru.eva.oriokslive.databinding.ActivityScheduleBinding
import ru.eva.oriokslive.ui.adapter.SchedulerPagerAdapter
import ru.eva.oriokslive.ui.adapter.SchedulerPagerAdapter.Companion.titles
import ru.eva.oriokslive.ui.base.BaseActivity
import ru.eva.oriokslive.ui.fragment.groups.GroupsFragment.Companion.EXTRA_GROUP
import ru.eva.oriokslive.utils.showToast

@AndroidEntryPoint
class SchedulerActivity : BaseActivity<ActivityScheduleBinding>() {
    override val bindingInflater: (LayoutInflater) -> ActivityScheduleBinding =
        ActivityScheduleBinding::inflate

    private val viewModel: ScheduleViewModel by viewModels()

    override fun setupUI() {
        setSupportActionBar(binding.toolbar)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
        }
        val group = intent.getStringExtra(EXTRA_GROUP) ?: ""
        val adapter = SchedulerPagerAdapter(this, group)

        viewModel.getSchedule(group)
        viewModel.scheduleExist.observe(this) {
            binding.viewPager.adapter = adapter
            TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
                tab.setText(titles[position])
            }.attach()
        }
        viewModel.onError.observe(this) { showToast(it) }
        viewModel.viewPagerPosition.observe(this) {
            binding.viewPager.currentItem = it
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.main, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem) = when (item.itemId) {
        R.id.action_date -> {
            viewModel.setViewPagerToCurrentWeek()
            true
        }
        else -> {
            onBackPressed()
            true
        }
    }
}
